contrib/platform/test/com/sun/jna/platform/win32/Advapi32Test.java

```diff
@@ -1005,6 +1005,200 @@ public void testImpersonateSelf() {
     	assertTrue(Advapi32.INSTANCE.RevertToSelf());
     }
 
+    public void testGetSetFileSecurityNoSACL() throws Exception {
+        int infoType = OWNER_SECURITY_INFORMATION
+                       | GROUP_SECURITY_INFORMATION
+                       | DACL_SECURITY_INFORMATION;
+
+        int memSize = 64 * 1024;
+        Memory memorySecurity = new Memory(memSize);
+        IntByReference sizeNeeded = new IntByReference(0);
+        // create a temp file
+        File file = createTempFile();
+        String filePath = file.getAbsolutePath();
+
+        try {
+
+            assertEquals("GetFileSecurity(" + filePath + ")", true,
+                         Advapi32.INSTANCE.GetFileSecurity(
+                                 filePath,
+                                 infoType,
+                                 memorySecurity,
+                                 memSize,
+                                 sizeNeeded));
+            assertEquals("SetFileSecurity(" + filePath + ")", true,
+                         Advapi32.INSTANCE.SetFileSecurity(
+                                 filePath,
+                                 infoType,
+                                 memorySecurity.getPointer(0)));
```

Comment:
This fails in my test and the fix is easy: just pass memorySecurity directly and don't indirect it.